label_studio/projects/functions/next_task.py

```diff
@@ -0,0 +1,256 @@
+
+        # WARNING! this call doesn't work after consequent annotate
+        num_tasks_with_current_predictions = task_with_current_predictions.count()
+        if cluster_num_solved_map:
+            task_with_current_predictions = task_with_current_predictions.annotate(
+                cluster_num_solved=Case(*cluster_num_solved_map, default=0, output_field=DecimalField())
+            )
+            # next task is chosen from least solved cluster and with lowest prediction score
+            possible_next_tasks = task_with_current_predictions.order_by('cluster_num_solved', 'predictions__score')
+        else:
+            possible_next_tasks = task_with_current_predictions.order_by('predictions__score')
+
+        num_annotators = project.annotators().count()
+        if num_annotators > 1 and num_tasks_with_current_predictions > 0:
+            # try to randomize tasks to avoid concurrent labeling between several annotators
+            next_task = _get_random_unlocked(
+                possible_next_tasks, user, upper_limit=min(num_annotators + 1, num_tasks_with_current_predictions)
+            )
+        else:
+            next_task = _get_first_unlocked(possible_next_tasks, user)
+    else:
+        # uncertainty sampling fallback: choose by random sampling
+        logger.debug(
+            f'Uncertainty sampling fallbacks to random sampling '
+            f'(current project.model_version={str(project.model_version)})'
+        )
+        next_task = _get_random_unlocked(tasks, user)
+    return next_task
+
+
+def get_not_solved_tasks_qs(user, project, prepared_tasks, assigned_flag):
+    if project.skip_queue == project.SkipQueue.REQUEUE_FOR_ME:
+        user_solved_tasks_array = user.annotations.filter(ground_truth=False, was_cancelled=False)
+    else:
+        user_solved_tasks_array = user.annotations.filter(ground_truth=False)
+    user_solved_tasks_array = (
+        user_solved_tasks_array.filter(task__isnull=False).distinct().values_list('task__pk', flat=True)
+    )
+    not_solved_tasks = prepared_tasks.exclude(pk__in=user_solved_tasks_array)
+
+    # if annotator is assigned for tasks, he must to solve it regardless of is_labeled=True
+    if not assigned_flag:
+        not_solved_tasks = not_solved_tasks.filter(is_labeled=False)
+
+    # show tasks with overlap > 1 first
+    if project.show_overlap_first:
+        # don't output anything - just filter tasks with overlap
+        logger.debug(f'User={user} tries overlap first from prepared tasks')
+        _, not_solved_tasks = _try_tasks_with_overlap(not_solved_tasks)
+        # queue_info += 'Show overlap first'
+
+    if project.skip_queue == project.SkipQueue.REQUEUE_FOR_ME:
+        # https://code.djangoproject.com/ticket/19726
+        if settings.DJANGO_DB == settings.DJANGO_DB_SQLITE:
+            not_solved_tasks = not_solved_tasks.order_by('annotations__was_cancelled', 'id')
```

Comment:
1. why do we use '-annotations__was_cancelled' for other dbs? 
2. why isn't this covered by tests and line 1?

Reply:
1. Added more info in the comment before this `if`
2. I think it's a problem of coverage collecting.